pkg/discovery/worker/dispatcher.go

```diff
@@ -63,44 +63,23 @@ func (d *Dispatcher) Init(c *ResultCollector) {
 
 // Register the k8sDiscoveryWorker and its monitoring workers
 func (d *Dispatcher) RegisterWorker(worker *k8sDiscoveryWorker) {
+	// Return the free worker to the pool
 	d.workerPool <- worker.taskChan
 }
 
-// Create Task objects for discovery and monitoring for each group of the nodes and pods
+// Create Task objects for discovery and monitoring for each node, and the pods and containers on that node
 // Dispatch the task to the pool, task will be picked by the k8sDiscoveryWorker
-// Receives the complete list of nodes in the cluster that are divided in groups and submitted as
-// Tasks to the DiscoveryWorkers to carry out the discovery of the pods, containers and resources
 func (d *Dispatcher) Dispatch(nodes []*api.Node, cluster *repository.ClusterSummary) int {
-
-	// make sure when len(node) < workerCount, worker will receive at most 1 node to discover
-	perTaskNodeLength := int(math.Ceil(float64(len(nodes)) / float64(d.config.workerCount)))
-	glog.V(3).Infof("The number of nodes per task is: %d", perTaskNodeLength)
-	assignedNodesCount := 0
-	assignedWorkerCount := 0
-	// Divide up the nodes into groups and assign each group to a separate task
-	for assignedNodesCount+perTaskNodeLength <= len(nodes) {
-		currNodes := nodes[assignedNodesCount : assignedNodesCount+perTaskNodeLength]
-
-		currPods := d.config.clusterInfoScraper.GetRunningAndReadyPodsOnNodes(currNodes)
-
-		currTask := task.NewTask().WithNodes(currNodes).WithPods(currPods).WithCluster(cluster)
-		d.assignTask(currTask)
-
-		assignedNodesCount += perTaskNodeLength
-
-		assignedWorkerCount++
-	}
-	if assignedNodesCount < len(nodes) {
-		currNodes := nodes[assignedNodesCount:]
-		currPods := d.config.clusterInfoScraper.GetRunningAndReadyPodsOnNodes(currNodes)
-		currTask := task.NewTask().WithNodes(currNodes).WithPods(currPods).WithCluster(cluster)
-		d.assignTask(currTask)
-
-		assignedWorkerCount++
-	}
-	glog.V(2).Infof("Dispatched discovery task to %d workers", assignedWorkerCount)
-
-	return assignedWorkerCount
+	go func() {
+		for _, node := range nodes {
+			currNodes := []*api.Node{node}
+			currPods := d.config.clusterInfoScraper.GetRunningAndReadyPodsOnNodes(currNodes)
```

Comment:
Nit: `Master` has a `GetRunningAndReadyPodsOnNode`. It might be worth having that simpler function  and using that here too.